### PR TITLE
feat(codemod): add codemod to upgrade Icon to Future Icon

### PR DESCRIPTION
**Pull request by HeartSquared**
## Why

[KZN-2677](https://cultureamp.atlassian.net/jira/software/c/projects/KZN/boards/634?selectedIssue=KZN-2677)

## What

README:
- [x] Add new codemod to list
- [x] Set up notes

CaMonogramIcon:
- [x] Update import to Brand
- [x] Update tag name to Brand
- [x] Add variant `enso`
- [x] If `inheritSize` - remove prop
- [x] If not `inheritSize` - add style to set size to 20px
- [x] All other props as is

SpinnerIcon:
- [x] Update import to LoadingSpinner
- [x] Update tag name to LoadingSpinner
- [x] Replace `aria-label` with `accessibilityLabel` or fallback of `"Loading"`
- [x] Remove `role`
- [x] Remove `viewBox`
- [x] All other props as is

LiveIcon:
- [x] Update to `sensors`
  - **TBC** with Jess whether we want to add to the default set
- Animated: OOS; keep it using legacy for now (but to later move to be a subcomponent of Tag)

Icon:
- [x] Update imports
- [x] Add `name` based on previous component name
- [x] Add `isFilled` if required based on previous component name
- [x] `role="presentation"` becomes `isPresentational`
- [x] `role="img"` to be removed (should have an `aria-label`)
- [x] `aria-label` becomes `alt`
- [x] `classNameOverride` becomes `className`
- [x] Leave `inheritSize` as is (expecting a TS error) and leave a comment above it
- [x] Change`aria-hidden` to `isPresentational`
- [x] Change `color` - inline style
- [x] Keep `data-testid` as is
- [x] Remove `fontSize` - doesn't do anything
- [x] Change `height` + `width` - convert to --icon-size
- [x] Keep `id` as is
- [x] Keep `onClick` as is
- [x] Remove `viewBox`

[KZN-2677]: https://cultureamp.atlassian.net/browse/KZN-2677?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/codemods/upgradeIconV1/getNewIconPropsFromOldIconName.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  ["InvisibleIcon", { name: "visibility_off" }],
+  ["ItalicsIcon", { name: "format_italic" }],
+  ["KebabIcon", { name: "more_vert" }],
+  ["KioskIcon", undefined],
+  ["KioskWhiteIcon", undefined],
+  ["LaunchIcon", { name: "play_circle" }],
+  ["LaunchWhiteIcon", undefined],
+  ["LeaderboardIcon", undefined],
+  ["ListViewIcon", { name: "list" }],
+  ["LiveIcon", { name: "sensors" }],
```
Comment (HeartSquared) [inline packages/components/codemods/upgradeIconV1/getNewIconPropsFromOldIconName.ts:154]: TBC with Jess whether we add this to the default set

**Review by dougmacknz — APPROVED** [review]
Awesome work

File: packages/components/codemods/utils/updateKaioImports.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        // Update import statement
+        statements[importIndex] = updatedImportDeclaration
+      })
+    }
+
+    return factory.updateSourceFile(rootNode, statements)
+  }
+
+/* Consumer helpers */
+export const setImportToRemove = (
```
Comment (mcwinter07) [inline packages/components/codemods/utils/updateKaioImports.ts:208]: I kind of understand these in the abstract by looking at how they're used in the upgradeIconV1 but can we add a little more context in the code comments, potentially with an example?

File: packages/components/codemods/upgradeIconV1/transformCaMonogramIconToBrand.spec.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+              <CaMonogramIcon role="presentation" />
+              <CaMonogramIcon role="img" aria-label="Add something" />
+            </>
+          )
+        `)
+      const outputAst = parseJsx(`
+          export const TestComponent = () => (
+            <>
+              <Brand variant="enso" style={{ width: "20px" }} role="presentation" />
+              <Brand variant="enso" style={{ width: "20px" }} role="img" aria-label="Add something" />
```
[the remark this answers is not in the document]
Comment (HeartSquared) [reply packages/components/codemods/upgradeIconV1/transformCaMonogramIconToBrand.spec.ts:56]: Oops! Yep you're right 😬 Will update 🙏🏻

File: packages/components/codemods/upgradeIconV1/getNewIconPropsFromOldIconName.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+  ["UserExitIcon", undefined],
+  ["UserIcon", { name: "person" }],
+  ["UserSelectIcon", { name: "person_check" }],
+  ["UserSettingsIcon", { name: "manage_accounts" }],
+  ["UsersIcon", { name: "group" }],
+  ["UserUpdateIcon", undefined],
+  ["VisibleIcon", { name: "visibility" }],
+  ["WritingIcon", undefined],
+  ["ZoomInIcon", { name: "zoom_in" }],
+  ["ZoomOutIcon", { name: "zoom_out" }],
```
[the remark this answers is not in the document]
Comment (HeartSquared) [reply packages/components/codemods/upgradeIconV1/getNewIconPropsFromOldIconName.ts:250]: Copy/paste is our best friend <3

**Pull request merged**